src/App/Fossa/LicenseScanner.hs

```diff
@@ -0,0 +1,119 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module App.Fossa.LicenseScanner (
+  licenseScanSourceUnit,
+  licenseNoScanSourceUnit,
+) where
+
+import App.Fossa.RunThemis (
+  execThemis,
+ )
+
+import Control.Carrier.Diagnostics qualified as Diag
+import Control.Carrier.StickyLogger (StickyLogger, logSticky)
+import Control.Effect.Lift
+
+import App.Fossa.ArchiveUploader
+import App.Fossa.FossaAPIV1 qualified as Fossa
+import Control.Effect.Diagnostics (Diagnostics, context, fatalText)
+import Control.Monad (unless)
+import Crypto.Hash (Digest, MD5, hash)
+import Data.ByteString qualified as B
+import Data.List (nub)
+import Effect.Exec (Exec, runExecIO)
+import Fossa.API.Types
+import Path
+import Srclib.Types (
+  LicenseSourceUnit (..),
+  LicenseUnit (..),
+  Locator (..),
+ )
+
+import Data.String.Conversion (encodeUtf8, toString, toText)
+import Data.Text (Text)
+
+import App.Fossa.EmbeddedBinary (ThemisBins, withThemisAndIndex)
+
+runLicenseScanOnDir ::
+  ( Has Diagnostics sig m
+  , Has (Lift IO) sig m
+  ) =>
+  Path Abs Dir ->
+  m [LicenseUnit]
+runLicenseScanOnDir scanDir = withThemisAndIndex $ themisRunner scanDir
+
+themisRunner ::
+  ( Has (Lift IO) sig m
+  , Has Diagnostics sig m
+  ) =>
+  Path Abs Dir ->
+  ThemisBins ->
+  m [LicenseUnit]
+themisRunner scanDir themisBins = do
+  runExecIO $ runThemis themisBins scanDir
+
+runThemis :: (Has Exec sig m, Has Diagnostics sig m) => ThemisBins -> Path Abs Dir -> m [LicenseUnit]
+runThemis themisBins scanDir = do
+  context "Running license scan binary" $ execThemis themisBins scanDir
+
+md5 :: B.ByteString -> Digest MD5
+md5 = hash
+
+scanAndUploadVendoredDep :: (Has Diag.Diagnostics sig m, Has (Lift IO) sig m, Has StickyLogger sig m) => ApiOpts -> Path Abs Dir -> VendoredDependency -> m Archive
```

Comment:
[Optional] [Prefer] I would collapse arguments with linebreak like you have for `themisRunner` for consistency.